src/pages/Detail/DetailContents/ProductInfo/ProductInfo.js

```diff
@@ -0,0 +1,105 @@
+import React, { useState } from 'react';
+import useFetch from '../../../../hooks/useFetch';
+import BookmarkBorderIcon from '@mui/icons-material/BookmarkBorder';
+import BookmarkIcon from '@mui/icons-material/Bookmark';
+import * as S from './ProductInfo.style';
+import { useNavigate, useParams } from 'react-router-dom';
+import { API } from '../../../../config';
+
+const ProductInfo = ({ info }) => {
+  const params = useParams();
+  const navigate = useNavigate();
+  const [isUpdateWishListFlag, setIsUpdateWishListFlag] = useState(false);
+  let addedWishlist = false;
+
+  const {
+    author_name,
+    description,
+    height,
+    item_name,
+    length,
+    materials,
+    production_year,
+    weight,
+    width,
+  } = info;
+
+  //FIXME - mock data devUrl
+  //const getAllWishList = '/data/wishlistOne.json';
+
+  const getAllWishList = `${API.WISHLIST}`;
+
+  //const interlockToWishList = `${API.WISHLIST}`;
+  const { loading, data } = useFetch(getAllWishList);
+
+  if (!loading) {
+    data.data.map(data => {
+      if (data.item_id === Number(params.id)) {
+        addedWishlist = true;
+        return;
+      }
+    });
+  }
+
+  const addItemToWishList = id => {
+    const addUrl = `${getAllWishList}/items/${id}`;
+    const params = {
+      method: 'POST',
+      headers: { 'Content-Type': 'application/json' },
+    };
+    fetch(addUrl, params).then(res => {
+      if (res.ok) {
+        addedWishlist = !addedWishlist;
+        setIsUpdateWishListFlag(prev => !prev);
+      } else throw new Error(res);
+    });
+    navigate(`/detail/${id}`);
+  };
+
+  const removeItemToWishList = id => {
+    //백엔드 API 준비중
+    /*
+     * 작성될 코드
+     */
+    navigate(`/detail/${id}`);
+  };
+
+  const controlWishList = id => {
+    if (addedWishlist) {
+      //위시리스트 아이템 삭제 API 준비중
+      removeItemToWishList(id);
+    } else {
+      addItemToWishList(id);
+    }
+  };
+
+  return (
+    <div>
+      <S.Title>작품명 : {item_name}</S.Title>
+      <S.Contents description={true}>작품 설명 : {description}</S.Contents>
+      <S.Contents>작가 : {author_name}</S.Contents>
+      <S.Contents>너비 : {width} cm</S.Contents>
+      <S.Contents>높이 : {length} cm</S.Contents>
+      <S.Contents>깊이 : {height} cm</S.Contents>
+      <S.Contents>무게 : {weight} kg</S.Contents>
+      <S.Contents>제작년도 : {production_year} 년</S.Contents>
+      <S.Contents>
+        소재 :{' '}
+        {materials.map((material, index) => {
+          if (materials[index] !== materials[materials.length - 1]) {
+            return (material = `${material}, `);
+          } else {
+            return material;
+          }
+        })}
```

Comment:
```suggestion
const artMaterials = materials.map((material, index) => {
          if (materials[index] !== materials[materials.length - 1]) {
            return (material = `${material}, `);
          } else {
            return material;
          }
        })

return(
<div>
     <S.Title>
     ....

       <S.Contents>
          소재 :{' '}
          {artMaterials}
        </S.Contents>
```

분리하면 JSX에서는 랜더된 결과만 보고, 로직은 따로 확인할 수 있어서 가독성이 좋을것같습니다.

Reply:
말씀해주신대로 JSX를 변수로 선언하였습니다.
가시성을 높이는데에 정말 좋은 방법인 것 같습니다.
감사합니다.